testsuite/formatter.py

```diff
@@ -8,10 +8,14 @@ class ChunkedJsonFormatter(PrettyJSONFormatter):
     """
     def __init__(self, a, b):
         super(ChunkedJsonFormatter, self).__init__(a, b)
-        self.snapshot = []
+        self.snapshot = {
```

Comment:
That sounds like a good idea! Mostly I just wanted to work out the data  flow with this PR, to figure out where to grab what we need.